### PR TITLE
Run R CMD check on v3.6, v4.0 and release

### LINKED ISSUE #986: R CMD check
Currently we are running `R CMD check` on R version 3.5, 3.6 and the latest release (currently 4.1) along with package versions from the time the respective R version was released. Keeping the backward compatibility with the v3.5 environment (with package versions from 2018) is increasingly proving to be difficult.

I've asked which R versions people are using in their GxP environments in our Slack channel. v3.5 only got a single vote so it's probably safe to "discard".

I would suggest that going forward we test on

- 3.6.3 with package versions from 2020-02-29
- 4.0.5 with package versions from 2021-03-31
- 4.1.x with latest package versions

Any thoughts @pharmaverse/admiral-devs

### PR DESCRIPTION
**Pull request by thomas-neitmann**
Close #986 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: .github/workflows/R-CMD-check.yml [hunk 1]
```diff
@@ -75,17 +65,10 @@ jobs:
           done < <(Rscript -e 'writeLines(remotes::system_requirements("ubuntu", "20.04"))')
 
       - name: Install dependencies
-        if: matrix.config.r == '3.6'
```
Comment (bms63) [inline .github/workflows/R-CMD-check.yml:78]: I can see why 3.5 is removed, but is 3.6 supposed to be removed from this as well?  I've never done this so all new to me

File: .github/workflows/R-CMD-check.yml [hunk 2]
```diff
@@ -44,17 +44,7 @@ jobs:
           http-user-agent: ${{ matrix.config.http-user-agent }}
           use-public-rspm: true
 
-      - uses: r-lib/actions/setup-pandoc@v1
-
-      - uses: actions/cache@v2
-        if: startsWith(runner.os, 'Linux') && matrix.config.r == '3.6'
```
Comment (bms63) [inline .github/workflows/R-CMD-check.yml:50]: okay for this to be removed?

**Review by bms63 — APPROVED** [review]

**Comment by thomas-neitmann** [comment]
@pharmaverse/admiral I finally managed to get the `R CMD check` working on 3.6, 4.0 and 4.1. I need someone to approve such that we can merge this. Once done we can finally get rid of workarounds we used in order to support v3.5.

File: R/globals.R [hunk 3]
```diff
@@ -84,14 +84,25 @@ globalVariables(c(
...
+  "temp_dose_freq",
+  "temp_new_dose_no",
+  "temp_num_of_doses",
+  "temp_dose_multiplier",
+  "temp_day_difference",
   "tmp_analysis_date",
   "tmp_dose_date",
   "tmp_seq_var",
   "EXDOSFRQ",
   "tmp_ldose_dt",
```
Comment (bms63) [inline R/globals.R:100]: Is this something to do for later - update to temp_* in other .R files?  Make a separate issue?

**Review by bms63 — APPROVED** [review]
So are all these changes related to the workflow and getting it to work properly?

**Comment by thomas-neitmann** [comment]
Yes, all related to getting the checks passed on all versions.

File: R/globals.R (hunk 3 above)
Comment (thomas-neitmann) [reply R/globals.R:100]: Yes, let's create a new issue to update this. I'd actually prefer `tmp_` over `temp_` but I guess either is fine as long as we are consistent.

**Pull request merged**